parley/src/lib.rs

```diff
@@ -124,6 +125,7 @@ pub use peniko::Font;
 
 pub use builder::{RangedBuilder, TreeBuilder};
 pub use context::LayoutContext;
+pub use cursor_test::CursorTest;
```

Comment:
What is the advantage of making this public?

Reply:
For writing tests for something like the text editor widget in Masonry. We can always stop exporting it in future versions if we find nobody uses it.

Reply:
I can't imagine a user story which would want this exported in its current form, but I can imagine that some of this code would be useful if made more modular.

I do think that the publicly exported "get the debug print of a cursor from the text and its corresponding layout" could be useful (where the layout is used only to find the soft line breaks in the text for debug printing) - caveated of course that it's a best effort. However, if we were actually doing that, we'd probably also want to properly use unicode_width, which I'm not sure we want to do. As it currently stands, this functionality is also tied in with the constructor creating the layout, which is pretty far from ideal.

The "quickly create a cursor from a needle in a string" is also useful to export publicly for people testing e.g. rich text, but again I think it's conflated with the other stuff.